assets/training/distillation/src/validate_pipeline.py

```diff
@@ -3,33 +3,335 @@
+            self._validate_data_generation_inputs()
+    
+    def _get_dataframe(self, file_path: str):
+        return pd.read_json(file_path, lines=True, chunksize=self._args.request_batch_size)
+    
+    def _get_inference_request_headers(self) -> dict:
+        key = self._args.teacher_model_endpoint_key
+        return {
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {key}"
+        }
+
+    def _validate_model_endpoint_args(self):
+        endpoint_name = self._args.teacher_model_endpoint_name
+        if endpoint_name:
+            endpoint_details = get_endpoint_details(mlclient_ws=self._mlclient, endpoint_name=endpoint_name)
+            self._args.teacher_model_endpoint_url = endpoint_details.get_endpoint_url()
+            self._args.teacher_model_endpoint_key = endpoint_details.get_endpoint_key()
+            model_asset_id = endpoint_details.get_deployed_model_id()
+            validate_teacher_model_details(model_asset_id)
+
+        if not self._args.teacher_model_endpoint_url \
+        or not self._args.teacher_model_endpoint_key:
+            raise ACFTValidationException._with_error(
+                AzureMLError.create(
+                    ACFTUserError,
+                    pii_safe_message=(
+                        "Endpoint URL and key are required fields for data generation."
+                    )
+                )
+            )
+    
+    def _validate_model_endpoint(self):
+        """Validates model endpoints availability by retrieving its details."""
+        base_url = get_base_url(self._args.teacher_model_endpoint_url)
+        request_headers = self._get_inference_request_headers()
+
+        # https://learn.microsoft.com/en-us/azure/machine-learning/reference-model-inference-info
+        response = requests.get(url=f"{base_url}/info", headers=request_headers)
+        response.raise_for_status()
+        response_data = response.json()
+        model_name = response_data.get("model_name")
+        logger.info(f"Model validated, model name - {model_name}")
+    
+    def _validate_model_inference(self):
+        """Validates a sample inference call.
+        
+        Raises:
+            HTTPError: If one occured.
+        """
+        # Prep data.
+        df = self._get_dataframe(file_path=self._args.train_file_path)
+        batch = next(df)
+        record = batch.iloc[0].to_dict()
+        
+        # Build inference payload
+        inference_params = {
+            MAX_NEW_TOKENS: self._args.teacher_model_max_new_tokens,
+            TEMPERATURE: self._args.teacher_model_temperature,
+            TOP_P: self._args.teacher_model_top_p,
+            **record
+        }
+
+        headers = self._get_inference_request_headers()
+        url = self._args.teacher_model_endpoint_url
+        response = requests.post(url=url, headers=headers, data=json.dumps(inference_params))
+        response.raise_for_status()
```

Comment:
could you please raise UserError here

Reply:
Thanks @novaturient95 for catch this! I'm supporting exponential backoff on failures. Please have a look now.